### PR TITLE
[chore] Update Coinbase Constructor Args Default

### PR DESCRIPTION
**Pull request by John-peterson-coinbase**
### What changed? Why?
- Update Coinbase constructor args so that passing undefined is valid. This will fallback to the full set of defaults.

### CODE REVIEW

File: src/coinbase/coinbase.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ export class Coinbase {
     useServerSigner = false,
     debugging = false,
     basePath = BASE_PATH,
-  }: CoinbaseOptions) {
+  }: CoinbaseOptions = {}) {
```
Comment (erdimaden) [inline src/coinbase/coinbase.ts:80]: that's great. I think we don't need to set empty string for `apiKeyName` and `privateKey`. let's make sure `CoinbaseOptions` is also updated: 

```
  /**
   * The API key name.
   */
  apiKeyName: string;

  /**
   * The private key associated with the API key.
   */
  privateKey: string;


```

Comment (John-peterson-coinbase) [reply src/coinbase/coinbase.ts:80]: If these are required fields than the deconstructor has to have default values for them So we can't really get around the empty string. 

Alternatively, if they are optional on the type and we do not provide the empty string default value -> we must do both undefined and empty string check which seems less clean

Comment (erdimaden) [reply src/coinbase/coinbase.ts:80]: I tested the changes locally and they worked for me. I removed the `?` mark from `apiKeyName` and `privateKey`, and also removed the default empty strings and default value for `CoinbaseOptions`. Let's sync on this to discuss further.

File: src/coinbase/coinbase.ts [hunk 2]
```diff
@@ -121,7 +121,7 @@ export class Coinbase {
     useServerSigner = false,
     debugging = false,
     basePath = BASE_PATH,
-  }: CoinbaseConfigureFromJsonOptions): Coinbase {
+  }: CoinbaseConfigureFromJsonOptions = {}): Coinbase {
```
[the remark this answers is not in the document]
Comment (John-peterson-coinbase) [reply src/coinbase/coinbase.ts:124]: this is for the configure from json options where passing undefined is valid

**Review by erdimaden — APPROVED** [review]

**Pull request merged**